test/nodes/test_multi_node_weighted_sampler.py

```diff
@@ -32,12 +30,11 @@ def setUp(self) -> None:
         }
         self.weights = {f"ds{i}": self._weights_fn(i) for i in range(self._num_datasets)}
 
-    def _setup_multi_node_wighted_sampler(self, num_samples, num_datasets, weights_fn, stop_criteria) -> Prefetcher:
+    def _setup_multi_node_wighted_sampler(self, num_samples, num_datasets, weights_fn, stop_criteria):
 
         datasets = {f"ds{i}": IterableWrapper(DummyIterableDataset(num_samples, f"ds{i}")) for i in range(num_datasets)}
         weights = {f"ds{i}": weights_fn(i) for i in range(num_datasets)}
-        node = MultiNodeWeightedSampler(datasets, weights, stop_criteria)
-        return Prefetcher(node, prefetch_factor=3)
```

Comment:
@andrewkho 

I moved the the epoch update logic inside mixer. The tests for the mixer nodes itself pass. 

But one thing thats not fitting in is when we add a Prefetcher node on top. Specifically, the Prefetcher node's state is None for the underlying node (`{'root': {'snapshot': None, 'steps_since_snapshot': 0}, 'num_yielded': 0}`) right after initialization (https://github.com/pytorch/data/blob/main/test/nodes/utils.py#L98). I think that is causing the epoch update logic to fail when we try to load it back here (https://github.com/pytorch/data/blob/main/test/nodes/utils.py#L136).